test/support/support.go

```diff
@@ -30,9 +30,10 @@ import (
 var (
 	ApplyOptions = metav1.ApplyOptions{FieldManager: "codeflare-test", Force: true}
 
-	TestTimeoutShort  = 1 * time.Minute
-	TestTimeoutMedium = 2 * time.Minute
-	TestTimeoutLong   = 5 * time.Minute
+	TestTimeoutShort           = 1 * time.Minute
+	TestTimeoutMedium          = 2 * time.Minute
+	TestTimeoutLong            = 5 * time.Minute
+	TestTimeoutGpuProvisioning = 30 * time.Minute
```

Comment:
Honestly, I am not a fan of configurable timeouts in general.
Default timeout should accommodate for all the used environments if possible.

About timeout generic naming - the problem I see is that every action with timeout is unique. By naming the timeouts generically you lose coupling between specific timeout. As a result if you need to adjust the timeout then you need to first search the code to see what timeout is used, also by raising shared timeout you also affect other tests, which may result in unintended longer test execution before fail.

Reply:
That's a great point, thanks Karel, it makes sense for unique cases to have their own timeout. I was thinking that since this new timeout is coincidently the max amount when running [e2e-tests locally](https://github.com/project-codeflare/codeflare-operator/blob/de7fc9dce224a92282bf33665465b3a15909b55a/Makefile#L364-L366) for example, it could be seen as just the max timeout and used generically. But, I understand what you mean as well.